packages/utils/src/database/methods/common.ts

```diff
@@ -1,26 +1,26 @@
 import { promisify } from 'util';
 
-import * as mongoDBLock from 'mongodb-lock';
 import * as mongoose from 'mongoose';
-import * as tri from 'tri';
 
 (mongoose.Promise as any) = global.Promise;
 
-import { debug as d } from '../../../utils/debug';
-import * as logger from '../../../utils/logging';
+import { debug as d } from '../../debug';
+import * as logger from '../../logging';
 
-const debug: debug.IDebugger = d(__filename);
-let cachedDb: mongoose.Connection;
-const lockName: string = 'index';
-const moduleName: string = 'Database:common';
+const mongoDBLock = require('mongodb-lock');
+const tri = require('tri');
+const debug = d(__filename);
+let cachedDb: mongoose.Connection | null = null;
+const lockName = 'index';
+const moduleName = 'Database:common';
 const { DatabaseConnection: dbConnectionString } = process.env; // eslint-disable-line no-process-env
 
 /**
  * Create a lock object.
  * @param {string} url - URL to lock in the database.
  */
 export const createLock = (url: string) => {
-    const collection = cachedDb.collection('locks');
+    const collection = cachedDb!.collection('locks');
```

Comment:
IIRC the connection could already exist and be shared and that's why we `connect` all the time, right?
Can you please explain that somewhere?

Also, why do we `connect` on `unlock` but not in `createLock`? Even if there's a reason I think that just for consistency we should do the same.

Reply:
I think @utsavized will be the right person to explain that somewhere because he was the one investigating this stuff.

about the `createLock`, you are always connecting before call to the `createLock` function.